node/src/accountant/test_utils.rs

```diff
@@ -92,53 +95,141 @@ impl Default for AccountantBuilder {
-        match self.payable_dao_factory {
-            None => {
-                self.payable_dao_factory =
-                    Some(PayableDaoFactoryMock::new().make_result(payable_dao))
-            }
-            Some(payable_dao_factory) => {
-                self.payable_dao_factory = Some(payable_dao_factory.make_result(payable_dao))
-            }
+impl<T> DaoDestination<T> {
+    fn num_rep(&self) -> u8 {
+        match self {
+            Self::AccountantDest(_) => 0,
+            Self::PayableScannerDest(_) => 1,
+            Self::ReceivableScannerDest(_) => 2,
+            Self::PendingPayableScannerDest(_) => 3,
+        }
+    }
+    fn inner_value(self) -> T {
+        match self {
+            Self::AccountantDest(dao) => dao,
+            Self::PayableScannerDest(dao) => dao,
+            Self::ReceivableScannerDest(dao) => dao,
+            Self::PendingPayableScannerDest(dao) => dao,
         }
-        self
     }
+}
 
-    pub fn receivable_dao(mut self, receivable_dao: ReceivableDaoMock) -> Self {
-        match self.receivable_dao_factory {
-            None => {
-                self.receivable_dao_factory =
-                    Some(ReceivableDaoFactoryMock::new().make_result(receivable_dao))
-            }
-            Some(receivable_dao_factory) => {
-                self.receivable_dao_factory =
-                    Some(receivable_dao_factory.make_result(receivable_dao))
+fn fill_vacancies_with_given_or_default_daos<const N: usize, T: Default>(
+    destination_matrix: [u8; N],
+    mut dao_set: Vec<DaoDestination<T>>,
+) -> Vec<Box<T>> {
+    let input_count = dao_set.len();
+    let (populated_queue, used_input) = destination_matrix.into_iter().fold(
+        (vec![], 0),
+        |(acc, used_input): (Vec<Box<T>>, usize), pos| {
+            if let Some(idx) = dao_set
+                .iter()
+                .position(|dao_std_pos| dao_std_pos.num_rep() == pos)
+            {
+                (
+                    plus(acc, Box::new(dao_set.remove(idx).inner_value())),
+                    used_input + 1,
+                )
+            } else {
+                (plus(acc, Box::new(Default::default())), used_input)
             }
-        }
-        self
+        },
+    );
+    if input_count != used_input {
+        panic!(
+            "you supplied DAO for unrealistic destination; look at the destination matrix that \
+             describes all proper usages of {:?} and decode those places by the num_rep() function \
+             pattern",
+            type_name::<T>()
+        )
     }
+    populated_queue
+}
 
-    pub fn pending_payable_dao(mut self, pending_payable_dao: PendingPayableDaoMock) -> Self {
-        match self.pending_payable_dao_factory {
-            None => {
-                self.pending_payable_dao_factory =
-                    Some(PendingPayableDaoFactoryMock::new().make_result(pending_payable_dao))
+macro_rules! update_factory {
```

Comment:
This is my personal opinion, I think we're making it complex to understand how the utility was built by using the macro, especially by using `destination_matrix`.

I'm not asking you to modify anything, a good thing to add will be if we explain by comments why those particular numbers inside the matrix.